src/jit/optimizer.cpp

```diff
@@ -7845,12 +7844,9 @@ GenTree *           Compiler::optIsBoolCond(GenTree *   condBranch,
     {
         isBool = true;
     }
-    else if (opr1->gtOper == GT_CNS_INT)
+    else if (opr1->IsIntegralConst(0) || opr1->IsIntegralConst(1))
```

Comment:
There seems to be another opportunity for `IsIntegralConst` a few lines above, 7834-7837.